mailpoet/mailpoet
# Coupon block error handling log errors for the newsletter stop sending for this newsletter [MAILPOET-4983]

## Description

This PR catches issues thrown by the Coupon block in the rendering/processing phase, logs, and notifies the user about it so that they can fix the issue and continue.


## QA notes

You'll need to make a newsletter with a coupon block run into a problem, one way to do this is adding a newsletter with a coupon block, but when on the `Send` page, just before clicking on the `Send` button, go to plugins' page and deactivate WooCommerce, this should prevent the email from being sent and make you end up with a listing like the attached screenshot.
https://d.pr/i/lZf3sV

## Linked tickets
[MAILPOET-4983]


[MAILPOET-4983]: https://mailpoet.atlassian.net/browse/MAILPOET-4983?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
